crev-data/src/proof/content.rs

```diff
@@ -0,0 +1,128 @@
+use crate::Url;
+use chrono::{self, prelude::*};
+use failure::bail;
+
+use self::super::review;
+use self::super::trust::TrustDraft;
+use crate::proof;
+use crate::proof::{Proof, ProofType};
+use crate::Result;
+
+pub type Date = chrono::DateTime<FixedOffset>;
+
+pub trait ContentCommon: std::fmt::Debug + std::string::ToString + ContentCommonClone {
+    fn as_any(&self) -> &dyn std::any::Any;
+
+    fn date(&self) -> &Date;
+    fn set_date(&mut self, date: &Date);
+    fn date_utc(&self) -> chrono::DateTime<Utc> {
+        self.date().with_timezone(&Utc)
+    }
+
+    fn author(&self) -> &crate::PubId;
+    fn set_author(&mut self, id: &crate::PubId);
+
+    fn author_id(&self) -> crate::Id {
+        self.author().id.clone()
+    }
+
+    fn author_url(&self) -> Url {
+        self.author().url.clone()
+    }
+
+    fn draft_title(&self) -> String;
+    fn validate_data(&self) -> Result<()>;
+
+    fn parse(s: &str) -> Result<Self>
```

Comment:
This just can't be here. One has to know the type, to be able to actually call parser to parse stuff into itself. 

I think we will have to have `struct ProofParser` that has a builder of some kind, where on can register bunch of stuff that implements `trait ContentParser`, and then call `fn parser(&self)` on that, and 

```
trait ContentParser {
   fn proof_type_name(&self) -> &'static str; /// the XXXX from ------ CREV XXXX PROOF ---- header, so that proof parser knows which proofs go to which content parsers
   fn parse(&self, &str) -> Box<dyn ContentCommon>;
}
```